compiler/parser.nim

```diff
@@ -1963,11 +1963,7 @@ proc parseObjectCase(p: var Parser): PNode =
   #|             | IND{=} objectBranches)
   result = newNodeP(nkRecCase, p)
   getTokNoInd(p)
-  var a = newNodeP(nkIdentDefs, p)
-  a.add(identWithPragma(p))
-  eat(p, tkColon)
-  a.add(parseTypeDesc(p))
-  a.add(p.emptyNode)
+  var a = parseIdentColonEquals(p, {withPragma})
```

Comment:
Can't suggest because it's outside the diff boundaries but the grammar update for this is

```diff
-objectCase = 'case' identWithPragma ':' typeDesc ':'? COMMENT?
+objectCase = 'case' declColonEquals ':'? COMMENT?
```

Then run `nim r compiler/parser`

Reply:
fixed https://github.com/nim-lang/Nim/pull/20480/commits/1c60bb9101cfeb8a6e6b479cb7a3798df66426ac